test/edge-integration.test.js

```diff
@@ -0,0 +1,230 @@
+    const testDomain = 'possibly-working-sawfish';
+    // Use the same package name as the existing working test
+    const packageName = 'Test';
+
+    const builder = await new CLI()
+      .prepare([
+        '--build',
+        '--plugin', resolve(__rootdir, 'src', 'index.js'),
+        '--verbose',
+        '--deploy',
+        '--target', 'c@e',
+        '--arch', 'edge',
+        '--compute-service-id', serviceID,
+        '--compute-test-domain', testDomain,
+        '--package.name', packageName,
+        '--package.params', 'HEY=ho',
+        '--package.params', 'ZIP=zap',
+        '--update-package', 'true',
+        '--fastly-gateway', 'deploy-test.anywhere.run',
+        '-p', 'FOO=bar',
+        '--fastly-service-id', '4u8SAdblhzzbXntBYCjhcK',
+        '--directory', testRoot,
+        '--entryFile', 'src/index.js',
+        '--bundler', 'webpack',
+        '--esm', 'false',
+      ]);
+    builder.cfg._logger = new TestLogger();
+
+    const res = await builder.run();
+    assert.ok(res, 'Fastly deployment should succeed');
+
+    return {
+      url: `https://${testDomain}.edgecompute.app`,
+      logger: builder.cfg._logger,
+    };
+  }
+
+  before(async function deployToBothPlatforms() {
+    this.timeout(600000); // 10 minutes for parallel deployment
+
+    // Fail explicitly if required credentials are missing
+    if (!process.env.HLX_FASTLY_AUTH) {
+      throw new Error('HLX_FASTLY_AUTH environment variable is required for Fastly integration tests. Please set it in GitHub repository secrets.');
+    }
+    if (!process.env.CLOUDFLARE_AUTH) {
+      throw new Error('CLOUDFLARE_AUTH environment variable is required for Cloudflare integration tests. Please set it in GitHub repository secrets.');
+    }
+
+    testRoot = await createTestRoot();
+    origPwd = process.cwd();
+
+    // Copy the edge-action fixture
+    await fse.copy(path.resolve(__rootdir, 'test', 'fixtures', 'edge-action'), testRoot);
+    process.chdir(testRoot);
+
+    // eslint-disable-next-line no-console
+    console.log('--: Starting parallel deployment to Cloudflare and Fastly...');
+
+    // Deploy to both platforms in parallel
+    const [cloudflareResult, fastlyResult] = await Promise.all([
+      deployToCloudflare(),
+      deployToFastly(),
+    ]);
+
+    deployments.cloudflare = cloudflareResult;
+    deployments.fastly = fastlyResult;
+
+    // eslint-disable-next-line no-console
+    console.log('--: Parallel deployment completed');
+    // eslint-disable-next-line no-console
+    console.log(`--: Cloudflare URL: ${deployments.cloudflare.url}`);
+    // eslint-disable-next-line no-console
+    console.log(`--: Fastly URL: ${deployments.fastly.url}`);
+  });
+
+  after(() => {
+    process.chdir(origPwd);
+  });
+
+  // Test suite that runs against both platforms
+  ['cloudflare', 'fastly'].forEach((platform) => {
+    describe(`${platform.charAt(0).toUpperCase() + platform.slice(1)} Platform`, () => {
```

Comment:
this should probably have the key word Integration in the test name, so that inclusions/exclusions work as expected